src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+        InternationalFixedChronology.DAY_OF_YEAR_RANGE.checkValidValue(dayOfYear, ChronoField.DAY_OF_YEAR);
+
+        if (dayOfYear == InternationalFixedChronology.DAYS_IN_YEAR + 1 && !InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear)) {
+            throw new DateTimeException("Invalid Year Day: " + prolepticYear + '/' + dayOfYear);
+        }
+
+        return new InternationalFixedDate(prolepticYear, dayOfYear);
+    }
+
+    /**
+     * Obtains a {@code InternationalFixedDate} representing a date in the International fixed calendar
+     * system from the epoch-day.
+     *
+     * @param epochDay the epoch day to convert based on 1970-01-01 (ISO)
+     * @return the date in International fixed calendar system, not null
+     * @throws DateTimeException if the epoch-day is out of range
+     */
+    static InternationalFixedDate ofEpochDay(final long epochDay) {
+        InternationalFixedChronology.EPOCH_DAY_RANGE.checkValidValue(epochDay, ChronoField.EPOCH_DAY);
+        long zeroDay = epochDay + InternationalFixedChronology.DAYS_0000_TO_1970;
+
+        if (zeroDay < 0) {
+            throw new DateTimeException("Invalid epoch: " + epochDay);
+        }
+
+        // The two values work great for any dates, just not the first (N/01/01) or the last of the year (N/0/0).
+        long year = (400 * zeroDay) / InternationalFixedChronology.DAYS_PER_CYCLE;
+        long doy = zeroDay - (InternationalFixedChronology.DAYS_IN_YEAR * year + InternationalFixedChronology.getLeapYearsBefore(year));
+
+        boolean isLeapYear = InternationalFixedChronology.INSTANCE.isLeapYear(year);
+
+        // In some cases, N/01/01 (January 1st) results in (N-1)/0/0, i.e. -1 day off.
+        if (doy == (InternationalFixedChronology.DAYS_IN_YEAR + 1) && !isLeapYear) {
+            year += 1;
+            doy = 1;
+        }
+
+        // In some cases, N/0/0 results in (N+1)/0/0 (rubbish), in a way +1 year off.
+        if (doy == 0) {
+            year -= 1;
+            doy = InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear ? 1 : 0);
+        }
+
+        return ofYearDay((int) year, (int) doy);
+    }
+
+    /**
+     * Consistency check for dates manipulations after calls to
+     *   {@link #plus(long, TemporalUnit)},
+     *   {@link #minus(long, TemporalUnit)},
+     *   {@link #until(AbstractDate, TemporalUnit)} or
+     *   {@link #with(TemporalField, long)}.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param month         the International fixed month, from -1 to 13 (-1 for Leap Day, 0 for Year Day)
+     * @param day           the International fixed day-of-month, from -1 to 28 (-1 for Leap Day, 0 for Year Day)
+     * @return
+     */
+    private static InternationalFixedDate resolvePreviousValid(final int prolepticYear, final int month, final int day) {
+        if (month == 0 && day == 0) {
```

Comment:
The problem is, `resolvePrevious(...)`/`resolvePreviousValid(...)` remember nothing about the previous state of a date - it doesn't actually know if you're moving _to_ a leap-day, or _away_ from it.  In such a situation, it's best to move from least granular (Biggest change) to most granular (smallest difference).
That is, when dealing with something like `with(ALIGNED_WEEK_OF_MONTH, ...)`, the first thing you'd need to change would be the month...
Actually, it would mean you wouldn't have to override `with(MONTH_OF_YEAR, ...)` for year-day, as the default behavior would work in that case.
The method is all about saying: "something was changed about this date, and now one (or more) of the values is off.  True it up".  You can't guarantee that day matches to month.